storage/aptosdb/src/lib.rs

```diff
@@ -1335,10 +1335,56 @@ impl DbWriter for AptosDB {
 
     fn get_state_snapshot_receiver(
         &self,
-        version: Version,
-        expected_root_hash: HashValue,
+        target_ledger_info: LedgerInfoWithSignatures,
+        target_output_with_proof: TransactionOutputListWithProof,
     ) -> Result<Box<dyn StateSnapshotReceiver<AccountStateBlob>>> {
         gauged_api("get_state_snapshot_receiver", || {
+            // Get the target version and expected root hash
```

Comment:
@msmouse -- this is the code that I mostly want to ensure is correct, i.e., once we've identified a version `V` to download all account states, we update the various data stores and start downloading all the states. Once all states are downloaded, the db should be in the correct state to move forward. From the smoke test, I don't see any complaints from the executor, but let me know if there's a data store or something that's missed 😄

Reply:
The added logic doesn't really belong here.. (it's more specific to state-sync)

At least give it a right name if you see the whole set of logic indeed belongs to the same function?

Is it that the helpers requires access to the stores so you have to do it on the DB side? How about refactor the restore_handler and use it from state sync? (I think the handler was created so that DB internal details are not exposed to the restore tooling)

Reply:
Also I don't quite understand what this is doing -- the txns are AT OR AFTER the state snapshot version, right? In my mind, we should do the snapshot first and then the ledger. Also strictly speaking, the txn, txn_out and events AT the version should be skipped, right? (you don't need them to start applying the next version)

Reply:
As discussed offline, I'll handle this by exposing an additional method in `DbWriter` that initializes all state once the accounts have all successfully synced 😄